packages/java/src/browser/java-client-contribution.ts

```diff
@@ -66,6 +63,7 @@ export class JavaClientContribution extends BaseLanguageClientContribution {
     protected onReady(languageClient: ILanguageClient): void {
         languageClient.onNotification(ActionableNotification.type, this.showActionableMessage.bind(this));
         languageClient.onNotification(StatusNotification.type, this.showStatusMessage.bind(this));
+        languageClient.onNotification(ProgressReportNotification.type, this.showProgressReportNotifications.bind(this));
```

Comment:
Interesting, I just read up on https://github.com/Microsoft/language-server-protocol/issues/70, which proposes a similar protocol extension. Tough it might be not only focused on languages. 

Is the JDT LS team aware of this it looks a bit competing, no?

Reply:
@JPinkney could you check the LSP issue? maybe that contains hits for the protocol, which we can add/incorporate for the Progress protocol in Theia. Mapping ProgressReportNotification should not be a big problem.

Reply:
@AlexTugarev While I think reporting progress on requests is a great addition, I think it's not necessarily the same thing: in jdt.ls, at least, background tasks are triggered on stuff that is not a reqest, think about a pom.xml being updated by a refactoring: the language server only gets notified with a "didChangeWatchedFiles()". As a user, I am interested in knowing when such things are going on.

Reply:
I should have referred to the PR https://github.com/Microsoft/vscode-languageserver-node/pull/261, which I consider to be the proposal for LSP#70. As far as I see they've removed the boundary of a request.